File: src/main/java/com/paperstreet/marketdata/ContractHandler.java
package com.paperstreet.marketdata;

import com.ib.client.Contract;
import com.paperstreet.utils.LogHandler;

import static com.paperstreet.marketdata.MarketDataConstants.*;

/**
 * Class for the creation of an IBKR contract to be traded.
 */
public class ContractHandler {

    private static final LogHandler logHandler = new LogHandler();

    /**
     * Converts the provided symbol to a valid IBKR contract. The contract helps identify and route
     * market data requests to the broker specifying the security type, exchange, and currency of US stocks.
     *
     * @param ticker The symbol to convert to a valid contract.
     * @return A contract associated with the symbol.
     */
    public static Contract contract(String ticker) {
        Contract contract = new Contract();
        contract.symbol(ticker);
        contract.secType(SECURITY_TYPE);
        contract.exchange(EXCHANGE);
        contract.currency(CURRENCY);

        logHandler.logInfo("Contract created for " + ticker);
        return contract;
    }

    public static Contract getContract(String symbol) {
        return contract(symbol);
    }
}

File: v2/marketdata/contract_handler.py
from ibapi.contract import Contract
from v2.connection_constants import *


class ContractHandler:
    @staticmethod
    def contract(ticker: str) -> Contract:
        """
        Converts the provided symbol to a valid IBKR contract.
        The contract helps identify and route market data requests to the broker
        specifying the security type, exchange, and currency of US stocks.
        """
        contract = Contract()
        contract.symbol = ticker
        contract.secType = SECURITY_TYPE
        contract.exchange = EXCHANGE
        contract.currency = CURRENCY

        print(f"Contract created for {ticker}")
        return contract

    @staticmethod
    def get_contract(symbol: str) -> Contract:
        """
        Returns an IBKR contract for the provided symbol.
        """
        return ContractHandler.contract(symbol)
